src/db/add_package.rs

```diff
@@ -237,7 +237,16 @@ fn get_optional_dependencies(pkg: &MetadataPackage) -> Vec<Feature> {
     pkg.dependencies
         .iter()
         .filter(|dep| dep.optional)
-        .map(|dep| Feature::new(dep.name.clone(), Vec::new()))
+        .map(|dep| {
+            let name = if let Some(rename) = &dep.rename {
+                rename.clone()
+            } else {
+                dep.name.clone()
+            };
+            let mut feature = Feature::new(name, Vec::new());
+            feature.set_optional_dependency(true);
```

Comment:
Rather than having this as a modifier, can you pass in an `optional_dependency` argument? That way it's not possible to forget to deal with it.

Reply:
Done